run/archiveloop

```diff
@@ -207,19 +218,34 @@ function archive_teslacam_clips () {
 
   /root/bin/connect-archive.sh
 
-  disconnect_usb_drives_from_host
+  #RG# Since we fix the files on script start, and use snapshot, don't disconnect
+  #disconnect_usb_drives_from_host
 
-  ensure_cam_file_is_mounted
+  #RG# take readonly snapshot to read from
+  log "Taking read-only snapshot of /backingfiles to /backingfiles/snap.."
+  btrfs subv snapshot -r /backingfiles /backingfiles/snap
+  
+  #RG# mount the snapshot instead
+  #ensure_cam_file_is_mounted
+  ensure_cam_snap_file_is_mounted
 
-  fix_errors_in_cam_file
```

Comment:
When a snapshot is taken of the filesystem holding the cam backingfile there's no guarantee that the contents of the backing file represent a valid file system, is there? I.e. the kernel could be writing metadata into the virtual file system, which could be partially committed to the backing file at the instant the snapshot is captured. It seems that an fsck should be performed on the FS contained within the snap before its contents are accessed.

Reply:
That's fine except the snap is read-only; I've done some simultaneous reading/writing and it seems to hold up. We could definitely add in fsck w/o repair and see how often there's a problem. Thoughts?